### PR TITLE
Added apply_transparency()

### LINKED ISSUE #6348: Transparency is not applied to RGBA palette when opening PNG
### What did you do?
I need to get a palette with transparency information from a PNG image. When saving this is correctly applied to the output image, when loading only an opaque palette is returned.

### What did you expect to happen?
I expect that image.info["transparency"] is applied to a returned RGBA palette, i.e. alpha cleared for a single transparent colour, or alpha replaced with the array of transparency values.

### What actually happened?
A completely opaque palette is returned

### What are your OS, Python and Pillow versions?

* OS: Windows 10
* Python: 3.9.5 (64 bit)
* Pillow: 9.1.1

```python
from PIL import Image

img = Image.new("P", (256, 256))
pixel = []
for y in range(256):
    for x in range(256):
        v = ((y // 16) & 1) ^ ((x//16) & 1)
        pixel.append(v)
img.putdata(pixel)
pal = []
for i in range(256):
    if i == 0:
        pal.extend([0,0,0])
    else:
        pal.extend([255,255,255])
img.putpalette(pal, "RGB")

img.save("test_A.png", transparency=0)
img.save("test_B.png", transparency=bytes([0] + [255] * 255))

pal = []
for i in range(256):
    if i == 0:
        pal.extend([0,0,0,0])
    else:
        pal.extend([255,255,255,255])
img.putpalette(pal, "RGBA")
img.save("test_C.png")

for f in ["test_A.png", "test_B.png", "test_C.png"]:
    img = Image.open(f)
    pal = img.getpalette("RGBA")
    print (img.info, img.mode, pal[0:8])
```
This sets the transparency in 3 different ways and saves images to the current directory, then loads them back. 
The images show the expected transparency if opened in a program like GIMP.

Output is:
```
{'transparency': 0} P [0, 0, 0, 255, 255, 255, 255, 255]
{'transparency': 0} P [0, 0, 0, 255, 255, 255, 255, 255]
{'transparency': 0} P [0, 0, 0, 255, 255, 255, 255, 255]
```
Expected output should be something like:
```
{'transparency': 0} P [0, 0, 0, 0, 255, 255, 255, 255]
{'transparency': 0} P [0, 0, 0, 0, 255, 255, 255, 255]
{'transparency': 0} P [0, 0, 0, 0, 255, 255, 255, 255]
```

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #6348

Adds `apply_transparency()`, a method to take a P mode image with a "transparency" key in the `im.info` dictionary, remove the "transparency" key and apply the transparency to the palette instead.

### CODE REVIEW

**Comment by hugovk** [comment]
Thanks, for a new API, let's add this to docs (an `automethod` in `docs/reference/Image.rst`?) and release notes.

**Comment by radarhere** [comment]
Ok, done

File: docs/reference/Image.rst [hunk 1]
```diff
@@ -202,6 +202,7 @@ This helps to get the bounding box coordinates of the input image:
 .. automethod:: PIL.Image.Image.getexif
 .. automethod:: PIL.Image.Image.getextrema
 .. automethod:: PIL.Image.Image.getpalette
+.. automethod:: PIL.Image.Image.apply_transparency
```
Comment (hugovk) [inline docs/reference/Image.rst:205]: The rest are alphabetical, shall we put this up near line 125?

**Comment by hugovk** [comment]
Thanks!

**Pull request merged**